test/remember.ts

```diff
@@ -0,0 +1,128 @@
+import * as assert from 'assert';
+
+import {
+  pipe,
+  subscribe,
+  of,
+  uponStart,
+  fromArray,
+  remember,
+  uponEnd,
+  merge,
+  makeSubject,
+} from '../src/index';
+
+describe('remember tests', () => {
+  it('should call hook only once with remember', done => {
+    let numHook = 0;
+    const of$ = pipe(
+      of('some value'),
+      uponStart(() => {
+        numHook++;
+      }),
+      remember()
+    );
+
+    let listener1Called = false;
+    let listener2Called = false;
+
+    pipe(
+      of$,
+      subscribe(
+        x => {
+          assert.strictEqual(x, 'some value');
+          listener1Called = true;
+        },
+        err => {
+          if (err) done(err);
+        }
+      )
+    );
+
+    pipe(
+      of$,
+      subscribe(
+        x => {
+          assert.strictEqual(x, 'some value');
+          listener2Called = true;
+        },
+        err => {
+          if (err) done(err);
+        }
+      )
+    );
+
+    setTimeout(() => {
+      assert.strictEqual(listener1Called, true, 'listener 1 called');
+      assert.strictEqual(listener2Called, true, 'listener 2 called');
+      assert.strictEqual(numHook, 1);
+      done();
+    }, 20);
+  });
+
+  it('should synchronously provide last value even after emission', done => {
+    const data$ = pipe(fromArray([1, 2, 3]), remember(2));
```

Comment:
Nice that `remember()` by default is 1 (just like xstream) but allows passing a number (unlike xstream)